dds/idl/marshal_generator.cpp

```diff
@@ -2268,7 +2268,7 @@ bool marshal_generator::gen_struct(AST_Structure* node,
         "  size_t size = 0;\n";
       std::ostringstream fields_encode;
       for (size_t i = 0; i < fields.size(); ++i) {
-        const unsigned id = be_global->get_id(node, fields[i], i);
+        const size_t id = be_global->get_id(node, fields[i], i);
```

Comment:
`get_id` returns `unsigned`, why do you want to change this?

Reply:
Some failed tests showed Warning of converting unsigned int to size_t on Line 2271 and Line 2390. Something is not right.

Reply:
It's probably complaining about the `i` argument. It would be safe to cast it to a `unsigned`.